### PR TITLE
KOGITO-2433 - BPMN Editor in VSCODE can not open file with missing SimulationAttributeSets

### PR DESCRIPTION
**Pull request by hasys**
Hi @romartin, @LuboTerifaj, @domhanak,

another small fix for third party editors better support. I provided fix for both Business Central and Kogito, since fix looks easy and safe for me. I provided tests for both Business Central and Kogito as well? Also I tested VS Code extension and Business Central and both loading provided file without any issues, so I think this PR is ready for review. Thank you!

Screenshot from 2020-06-17 13-15-09
Screenshot from 2020-06-17 14-10-44

VSIX can be downloaded here: https://drive.google.com/file/d/1Ab4dlZued6NFlgWlz0OR9WPd2FAbNKRz/view?usp=sharing
Business Central can be downloaded here:  https://drive.google.com/file/d/1tdHWSgJzw09ciYvUyM_B2coUrp69-q1_/view?usp=sharing

### CODE REVIEW

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Hi @domhanak, @LuboTerifaj, @romartin 


test fails in Full downstream build doesn't related to this PR:

<details>
  <summary>Failed tests</summary>
  <pre>

org.kie.processmigration.it.ProcessMigrationIntegrationTest.testReadinessCheck
--------------------------------------------------------------------------------------------------------------------------------------

Error Message

Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)

Stacktrace

org.apache.http.conn.HttpHostConnectException: Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testReadinessCheck(ProcessMigrationIntegrationTest.java:114)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testReadinessCheck(ProcessMigrationIntegrationTest.java:114)

org.kie.processmigration.it.ProcessMigrationIntegrationTest.testMigration
--------------------------------------------------------------------------------------------------------------------------------------

Error Message

Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)

Stacktrace

org.apache.http.conn.HttpHostConnectException: Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.createPlan(ProcessMigrationIntegrationTest.java:179)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.createMigration(ProcessMigrationIntegrationTest.java:148)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testMigration(ProcessMigrationIntegrationTest.java:134)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.createPlan(ProcessMigrationIntegrationTest.java:179)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.createMigration(ProcessMigrationIntegrationTest.java:148)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testMigration(ProcessMigrationIntegrationTest.java:134)

org.kie.processmigration.it.ProcessMigrationIntegrationTest.testLivenessCheck
-------------------------------------------------------------------------------------------------------------------------------------

Error Message

Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)

Stacktrace

org.apache.http.conn.HttpHostConnectException: Connect to localhost:8180 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testLivenessCheck(ProcessMigrationIntegrationTest.java:108)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at org.kie.processmigration.it.ProcessMigrationIntegrationTest.testLivenessCheck(ProcessMigrationIntegrationTest.java:108)

</pre>
</details>

PR is ready for review. Thank you!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/AssignmentsInfosTest.java [hunk 1]
```diff
@@ -21,11 +21,14 @@
...
 import org.eclipse.bpmn2.DataOutput;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.customproperties.ParsedAssignmentsInfo;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/AssignmentsInfosTest.java:31]: why need for running with mockito?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/BoundaryEventPropertyReaderTest.java [hunk 2]
```diff
@@ -29,15 +29,18 @@
...
 import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.TestUtils.mockBounds;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.TestUtils.mockFeatureMapEntry;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/BoundaryEventPropertyReaderTest.java:43]: No need for declaring it again, the annotation is `inherited` from superclasses

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/CallActivityPropertyReaderTest.java [hunk 3]
```diff
@@ -24,15 +24,18 @@
...
 import org.kie.workbench.common.stunner.bpmn.backend.converters.tostunner.DefinitionResolver;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.di;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/CallActivityPropertyReaderTest.java:38]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/CatchEventPropertyReaderTest.java [hunk 4]
```diff
@@ -23,10 +23,13 @@
 import org.eclipse.bpmn2.Event;
 import org.eclipse.bpmn2.EventDefinition;
 import org.eclipse.bpmn2.SignalEventDefinition;
+import org.junit.runner.RunWith;
 import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.mockito.Mockito.when;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/CatchEventPropertyReaderTest.java:32]: no need for this, inherited

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/DefinitionsPropertyReaderTest.java [hunk 5]
```diff
@@ -25,14 +25,17 @@
...
 import org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.properties.util.PropertyWriterUtils;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.tostunner.DefinitionResolver;
 import org.kie.workbench.common.stunner.bpmn.definition.property.diagram.imports.WSDLImport;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.di;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/DefinitionsPropertyReaderTest.java:38]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/EmbeddedSubprocessPropertyReaderTest.java [hunk 6]
```diff
@@ -23,13 +23,16 @@
...
+import org.junit.runner.RunWith;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.customproperties.CustomElement;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.tostunner.DefinitionResolver;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertTrue;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.di;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/EmbeddedSubprocessPropertyReaderTest.java:35]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/EventSubprocessPropertyReaderTest.java [hunk 7]
```diff
@@ -23,13 +23,16 @@
...
+import org.junit.runner.RunWith;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.customproperties.CustomElement;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.tostunner.DefinitionResolver;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertTrue;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.di;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/EventSubprocessPropertyReaderTest.java:35]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/InputAssignmentReaderTest.java [hunk 8]
```diff
@@ -22,11 +22,14 @@
...
 import org.eclipse.bpmn2.FormalExpression;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.kie.workbench.common.stunner.bpmn.backend.converters.customproperties.AssociationDeclaration;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.kie.workbench.common.stunner.bpmn.backend.converters.fromstunner.Factories.bpmn2;
 
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/InputAssignmentReaderTest.java:32]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/SimulationAttributeSetsTest.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+import bpsim.TimeParameters;
+import bpsim.impl.BpsimFactoryImpl;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.stunner.bpmn.definition.property.simulation.SimulationAttributeSet;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/SimulationAttributeSetsTest.java:30]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/compatibility/EclipseImportTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.stunner.bpmn.backend.service.diagram.marshalling.BPMNDiagramMarshallerBaseTest;
+import org.kie.workbench.common.stunner.core.diagram.Diagram;
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+import org.kie.workbench.common.stunner.core.graph.Graph;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/compatibility/EclipseImportTest.java:30]: no need for this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/test/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/tostunner/properties/SimulationAttributeSetsTest.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+import bpsim.TimeParameters;
+import bpsim.impl.BpsimFactoryImpl;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.stunner.bpmn.definition.property.simulation.SimulationAttributeSet;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/test/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/tostunner/properties/SimulationAttributeSetsTest.java:30]: no need for this

**Comment by hasys** [comment]
Hi @romartin,

About `@RunWith` annotation, I had an experience that tests without this annotation were not executed. So one day a lot of tests can stop working and we won't know about it since we doesn't control amount of running tests or something like this, so I think it is good to have this annotation. It doesn't cost us much and can prevent issues in the future, so I think it is good to have `@RunWith` annotation for all test classes.

About `bpmn2` extension, you are right it has nothing with real cause of not loaded BPMN file, in fact I had to rename it to `bpmn` to test it in Business Central as you pointed out.

The issue was in NullPointer exception when some Simulation Attributes missed in the file. So the fix was check for missing Simulation Attributes before process them. And it is applies for both marshallers backend and client, fix is pretty safe (if Simulation Attributes are null, return empty Simulation Attributes node instead of throwing NullPointer exception) so I applied it to both Backend and Client side marshallers.

Thank you @romartin for good questions, and let me know if what I wrote make sense to you.

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute compile downstream build

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Pull request merged**